src/main/java/io/cdap/plugin/gcp/firestore/common/FirestoreConfig.java

```diff
@@ -129,6 +140,15 @@ public void validate(FailureCollector collector) {
     IdUtils.validateReferenceName(referenceName, collector);
```

Comment:
Similarly `validateDatabaseName(collector)` should be called here instead of separately calling it in sub classes.

Reply:
Subclasses are expected to call `super.validate(collector)`.

Reply:
Got it. Fixed!